moped-editor/src/views/projects/projectView/ProjectComponents/ProjectComponentsList.js

```diff
@@ -27,20 +29,22 @@ const ProjectComponentsList = ({
 
   const onListItemClick = (component) => {
     setIsClickedComponentRelated(false);
-    // If we are editing a component, clicking on a component should not change the clicked component
+    // Clear clickedComponent and draftEditComponent when we are not selecting for edit
     if (isExpanded(component)) {
       setClickedComponent(null);
+      editDispatch({ type: "clear_draft_component" });
     } else if (isNotCreatingOrEditing) {
       setClickedComponent(component);
+      editDispatch({ type: "set_draft_component", payload: component });
     }
   };
 
-  const onStartEditingComponent = (component) => {
-    if (isSignalComponent(component)) {
-      editDispatch({ type: "start_attributes_edit" });
-    } else {
-      editDispatch({ type: "start_edit", payload: component });
-    }
+  const onEditAttributes = () =>
+    editDispatch({ type: "start_attributes_edit" });
+
+  const onEditMap = (component) => {
+    editDispatch({ type: "start_map_edit", payload: component });
+    onEditFeatures();
```

Comment:
These actions used to be handled when the **Attributes** or **Map** buttons were clicked in the extra dialog that is being removed in this PR.